### DISCUSSION TITLE
SQL Server Encryption Issue

### DISCUSSION THREAD

**Discussion by EthanKC** [nextgenhealthcare/connect#6199] [Q&A] [answered]
I'm working with an organization on a Mirth implementation. I have access to a server that I've installed Mirth on for them. They have set up a Microsoft SQL Server Database on a different server that I don't have access to. I've been able to successfully connect Mirth to this SQL Server DB previously, but now they are needing to turn on forced encryption for security purposes. The problem is, whenever they turn on forced encryption, they are getting this error: 
“Encryption is required to connect to this server but the client library does not support encryption. Please upgrade your client library. [CLIENT: IP of Mirth Server]”

I did some research, and added "encrypt=true;trustServerCertificate=true" to the connection string, but that's not working. Here is the full string:

jdbc:jtds:sqlserver://SERVERNAME:PORT/DBNAME;encrypt=true;trustServerCertificate=true

Any idea what I'm doing wrong, and what I can do to get this working?

**Comment by pacmano1** [comment]
Wrong driver.

`jdbc:sqlserver://`

oops.. one sec.  read too fast... lol.   Let me fire up one of my test machines.

**Reply by ab-mg-23** [reply, marked as answer]
I think you are on the right track,, the URL is for the jTDS driver but using the MS Driver options. The equivalent with jTDS for `encrypt=true;trustServerCertificate=true` should be `ssl=require`. For the driver to authenticate the certificate presented by the server would be `ssl=authenticate`.

**Reply by pacmano1** [reply]
Maybe I should look at my own repos, lol.

https://github.com/pacmano1/Mirth-Snippets/blob/main/mssql_jdbc_connect_strings.txt

**Reply by EthanKC** [reply]
Ok, so I'd need to use this:

jdbc:jtds:sqlserver://SERVERNAME:PORT/DBNAME;ssl=require

And they would need to make sure their certificate is using ssl=authenticate?

**Reply by EthanKC** [reply]
Side note: is there a good resource I can use to better understand these strings and how they work? I pretty much got what I have through google and trial and error, but I don't fully understand why the string I'm using does or doesn't work.

**Reply by pacmano1** [reply]
https://learn.microsoft.com/en-us/sql/connect/jdbc/setting-the-connection-properties?view=sql-server-ver15#properties

**Reply by ab-mg-23** [reply]
And for completeness, the link to the jTDS URL properties. https://jtds.sourceforge.net/faq.html#urlFormat. I would strongly recommend switching to using the MS JDBC Driver since that is actively supported.

**Comment by pacmano1** [comment]
I threw this together with some chatGPT help mostly because adding a TLS cert to MSSQL was a tad more complicated than I thought it was due to its requirements for the certificate.

https://github.com/pacmano1/Mirth-Snippets/blob/main/mirth_and_MSSQL_withTLS.md
